src/color-chart/color-chart.js

```diff
@@ -43,11 +43,15 @@ const Self = class ColorChart extends NudeElement {
 
 	connectedCallback () {
 		super.connectedCallback();
-		customElements.whenDefined("color-scale").then(() => this.render());
+		this._el.chart.addEventListener("colorschange", this, {capture: true});
+	}
+
+	disconnectedCallback () {
+		this._el.chart.removeEventListener("colorschange", this, {capture: true});
 	}
 
 	handleEvent (evt) {
-		if (evt.target.tagName === "COLOR-SCALE" && evt.name === "computedColors") {
+		if (evt.target.tagName === "COLOR-SCALE" && evt.name === "colors") {
```

Comment:
> Why? If the colors actually changed it should propagate to computedColors, no?

The change to `colors` does propagate to `computedColors`; however, the `colorschange` event is fired when the `colors` prop is changed (and `evt.name` is equal to `colors`):

```js
static events = {
	colorschange: {
		propchange: "colors",
	},
};
```

We can definitely change it to (in that case, `evt.name` will be `computedColors`):

```js
static events = {
	colorschange: {
		propchange: "computedColors",
	},
};
```

But since we didn't expose any events previously, I thought it's worth exposing the `colorschange` event and making the event and the corresponding property names match. That was my thought process when making this change.

Another option would be also to expose the `computedcolorschange` event (and listen to it instead):

```js
static events = {
	colorschange: {
		propchange: "colors",
	},
	computedcolorschange: {
		propchange: "computedColors",
	},
};
```

I'm a bit concerned, though, about whether we need it: either of the props is easily accessed if either of the events is listened to. And `colorschange` is shorter. 🙃

Does my explanation make sense?

Reply:
As a user, you want the `colorschange` event to fire when the colors change. If the `colors` attribute changes and `computedColors` _don't_ change, that's because the colors didn't _actually_ change. So I would suggest simply changing when the event fires. Calling it `computedcolorschange` is both awkwardly long, named around implementation, and makes the simple/common case harder to enable the complex/rare case.